pennylane/transforms/commutation_dag.py

```diff
@@ -0,0 +1,1049 @@
+        "WireCut": [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+        "RX": [0, 1, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 1, 0, 0, 0, 1, 0, 0, 0, 0],
+        "RY": [0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0],
+        "RZ": [0, 0, 0, 1, 0, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 1, 1, 1, 0, 0, 0, 0, 1, 0, 0],
+        "PhaseShift": [0, 0, 0, 1, 0, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 1, 1, 1, 0, 0, 0, 0, 1, 0, 0],
+        "Rot": [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0],
+        "MultiRZ": [0, 0, 0, 1, 0, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 1, 1, 1, 0, 0, 0, 0, 1, 0, 0],
+        "Identity": [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0],
+        "U1": [0, 0, 0, 1, 0, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 1, 1, 1, 0, 0, 0, 0, 1, 0, 0],
+        "U2": [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0],
+        "U3": [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0],
+        "IsingXX": [0, 1, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 1, 0, 0, 0, 1, 0, 0, 0, 0],
+        "IsingYY": [0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0],
+        "IsingZZ": [0, 0, 0, 1, 0, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 1, 1, 0, 1, 1, 1, 0, 0, 0, 0, 1, 0, 0],
+        "QubitStateVector": [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+        "BasisState": [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0],
+    }
+)
+"""OrderedDict[str, list[int]]: Represents the commutation relations between each gate. Positions in the array are
+the one defined by the position dictionary. 1 represents commutation and 0 non-commutation."""
+
+# fmt: on
+
+
+position = dict(zip(commutation_map, range(len(commutation_map))))
+"""OrderedDict[str, int]: represents the index of each gates in the list of the commutation_map dictionary."""
+
+
+def intersection(wires1, wires2):
+    r"""Check if two sets of wires intersect.
+
+    Args:
+        wires1 (pennylane.wires.Wires): First set of wires.
+        wires2 (pennylane.wires.Wires: Second set of wires.
+
+    Returns:
+         bool: True if the two sets of wires are not disjoint and False if disjoint.
+    """
+    return len(qml.wires.Wires.shared_wires([wires1, wires2])) != 0
+
+
+def simplify_rotation(rot):
+    r"""Simplify a general one qubit rotation into RX, RY and RZ rotation.
+
+    Args:
+        rot (pennylane.Rot): One qubit rotation.
+
+    Returns:
+         qml.operation: Simplified rotation if possible.
+    """
+
+    if np.allclose(np.mod(rot.data[0], 2 * np.pi), np.pi / 2) and np.allclose(
+        np.mod(rot.data[2], -2 * np.pi), -np.pi / 2
+    ):
```

Comment:
@rmoyard I just realized that a lot of the simplification functions could be made more efficient by simply taking the `mod` once, at the beginning of the functions:

```python
    wires = wires
    params = op.parameters

    p0, p1, p2 = np.mod(params, 2 * np.pi)

    if np.allclose(p0, np.pi / 2) and np.allclose(np.mod(params[2], -2 * np.pi), -np.pi / 2):
        return qml.RX(params[1], wires=wires)

    if np.allclose(p0, 0) and np.allclose(p2, 0):
        return qml.RY(params[1], wires=wires)

    if np.allclose(p1, 0):
        return qml.RZ(params[0] + params[2], wires=wires)

    if np.allclose(p0, np.pi) and np.allclose(p1, np.pi / 2) and np.allclose(p2, 0):
        return qml.Hadamard(wires=wires)

    return op
```

Worth considering!

Reply:
Good ideas @josh146, I will add those modifications!